<a href="https://colab.research.google.com/github/Tirtayuda12/Machine_Learning-Telu/blob/main/Week%205/Google%20Colab/Week_5_EDA_on_IT_salary_EU_report_2020.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

NAMA : Tirtayuda Munggarana

NIM : 1103202108

Kelas : TK-44-G04

### Langkah 1 : sambungkan google colab dengan google drive untuk mengakses dataset pada google drive

prompt chatGPT : buatkan kode untuk menyambungkan google colab dengan google drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
import pandas as pd
# path ke dataset
file_path = "/content/drive/MyDrive/DatasetML/IT Salary Survey EU  2020.csv"

# membaca sebagai dataframe
df = pd.read_csv(file_path)

## Exploratory Data Analysis (EDA)

prompt chatGPT : berikan petunjuk atau langkah dalam melakukan exploratory data analysis
ChatGPT


Exploratory Data Analysis (EDA) adalah proses analisis awal yang dilakukan pada dataset untuk memahami karakteristik data, mengidentifikasi pola, menemukan anomali, dan merumuskan hipotesis yang dapat diuji lebih lanjut. Berikut adalah langkah-langkah umum dalam melakukan EDA:

1. Memahami Dataset:
  * Baca Dokumentasi: Pelajari informasi tentang dataset, termasuk penjelasan setiap kolom (fitur) dan nilai-nilai yang mungkin.
  * Identifikasi Jenis Data: Tentukan tipe data untuk setiap kolom (numerik, kategori, string, dll.).

2. Pemeriksaan Data Awal:
  * Tampilkan Data: Lihat beberapa baris pertama dataset untuk memahami struktur dan format data.
  * Periksa Informasi: Gunakan df.info() (dengan df adalah DataFrame Pandas) untuk melihat informasi tentang jumlah entri, tipe data, dan apakah ada nilai yang hilang.
3. Membersihkan Data:
  * Periksa dan Tangani Nilai Kosong: Identifikasi dan tangani nilai kosong atau null. Anda bisa menghapus baris atau mengisi nilai yang hilang dengan metode seperti interpolasi atau imputasi.
  * Penanganan Anomali: Identifikasi dan tangani nilai yang tidak mungkin atau anomali dalam data.
4. Analisis Statistik Deskriptif:
  * Ringkasan Statistik: Gunakan df.describe() untuk melihat statistik deskriptif seperti mean, median, kuartil, dll. untuk kolom-kolom numerik.
  * Distribusi Data: Visualisasikan distribusi data menggunakan histogram, boxplot, atau KDE plot.
5. Explorasi Variabel:
  * Univariat Analysis: Analisis variabel secara individual untuk memahami distribusi dan pola data.
  * Bivariat Analysis: Analisis hubungan antara dua variabel. Gunakan scatter plot untuk melihat korelasi antar variabel numerik dan boxplot untuk membandingkan distribusi berdasarkan kategori.
6. Visualisasi Data:
  * Histogram: Untuk melihat distribusi variabel numerik.
  * Boxplot: Untuk melihat distribusi, kuartil, dan outlier.
  * Barplot: Untuk melihat frekuensi atau perbandingan antara kategori.
  * Scatter plot: Untuk melihat hubungan antar variabel numerik.
7. Pengelompokan dan Agregasi:
  * Pivot Tables: Untuk mengelompokkan data dan melihat statistik agregat berdasarkan beberapa fitur.
  * Groupby: Untuk melakukan pengelompokan data berdasarkan kategori tertentu dan menghitung statistik di setiap grup.
8. Analisis Korelasi:
  * Heatmap: Untuk visualisasi matriks korelasi antar variabel numerik.
  * Pairplot: Untuk melihat scatter plot dan distribusi variabel secara bersamaan.
9. Penyajian Informasi:
  * Gunakan narasi dan visualisasi yang jelas untuk menyajikan temuan dan insight dari analisis data.
  * Buat kesimpulan dan hipotesis yang dapat diuji lebih lanjut.
10. Iterasi dan Pemrosesan Ulang:
  * Setelah melakukan EDA awal, mungkin Anda perlu kembali ke langkah-langkah sebelumnya, melakukan pembersihan tambahan, atau mengeksplorasi aspek data yang lebih dalam.
  
  EDA adalah langkah awal yang penting sebelum membangun model atau melakukan analisis lanjutan. Ini membantu memahami data secara menyeluruh dan membimbing pengambilan keputusan yang tepat dalam analisis data.

### Langkah 1

In [ ]:
df.head()

Timestamp   Age Gender    City           Position   \
0  24/11/2020 11:14:15  26.0   Male  Munich   Software Engineer   
1  24/11/2020 11:14:16  26.0   Male  Berlin   Backend Developer   
2  24/11/2020 11:14:21  29.0   Male  Berlin   Software Engineer   
3  24/11/2020 11:15:24  28.0   Male  Berlin  Frontend Developer   
4  24/11/2020 11:15:46  37.0   Male  Berlin   Backend Developer   

  Total years of experience Years of experience in Germany Seniority level  \
0                         5                              3          Senior   
1                         7                              4          Senior   
2                        12                              6            Lead   
3                         4                              1          Junior   
4                        17                              6          Senior   

  Your main technology / programming language  \
0                                  TypeScript   
1                                        Ruby   
2                     Javascript / Typescript   
3                                  Javascript   
4                                     C# .NET   

  Other technologies/programming languages you use often  ...  \
0                    Kotlin, Javascript / Typescript      ...   
1                                                NaN      ...   
2                    Javascript / Typescript, Docker      ...   
3                                                NaN      ...   
4                             .NET, SQL, AWS, Docker      ...   

   Annual bonus+stocks one year ago. Only answer if staying in same country  \
0                                              10000                          
1                                               5000                          
2                                             100000                          
3                                                NaN                          
4                                                NaN                          

  Number of vacation days           Employment status   Сontract duration  \
0                      30          Full-time employee  Unlimited contract   
1                      28          Full-time employee  Unlimited contract   
2                      30  Self-employed (freelancer)  Temporary contract   
3                      24          Full-time employee  Unlimited contract   
4                      29          Full-time employee  Unlimited contract   

  Main language at work Company size Company type  \
0               English       51-100      Product   
1               English     101-1000      Product   
2               English     101-1000      Product   
3               English       51-100      Startup   
4               English     101-1000      Product   

  Have you lost your job due to the coronavirus outbreak?  \
0                                                 No        
1                                                 No        
2                                                Yes        
3                                                 No        
4                                                 No        

  Have you been forced to have a shorter working week (Kurzarbeit)? If yes, how many hours per week  \
0                                                NaN                                                  
1                                                NaN                                                  
2                                                NaN                                                  
3                                                NaN                                                  
4                                                NaN                                                  

  Have you received additional monetary support from your employer due to Work From Home? If yes, how much in 2020 in EUR  
0                                                NaN                                    

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1253 entries, 0 to 1252
Data columns (total 23 columns):
 #   Column                                                                                                                   Non-Null Count  Dtype  
---  ------                                                                                                                   --------------  -----  
 0   Timestamp                                                                                                                1253 non-null   object 
 1   Age                                                                                                                      1226 non-null   float64
 2   Gender                                                                                                                   1243 non-null   object 
 3   City                                                                                                                     1253 non-null   o

In [ ]:
print(df.dtypes)

Timestamp                                                                                                                   object
Age                                                                                                                        float64
Gender                                                                                                                      object
City                                                                                                                        object
Position                                                                                                                    object
Total years of experience                                                                                                   object
Years of experience in Germany                                                                                              object
Seniority level                                                                    

Setelah dilihat menggunakan method info, ada sebanyak 1253 baris dan 22 kolom pada dataset IT salary EU report 2020.

Tiap kolom juga terdapat banyak kekeliruan tipe data contohnya tipe data pada kolom total years of experience yang harusnya numerik namun setelah di cek bertipe data objek

In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Statistik deskriptif
print(df.describe())

# Memeriksa missing values
print(df.isnull().sum())

# Visualisasi distribusi gaji
plt.figure(figsize=(10, 6))
sns.histplot(df['Salary'], bins=30, kde=True, color='skyblue')
plt.title('Distribution of IT Salaries in EU 2020')
plt.xlabel('Salary (EUR)')
plt.ylabel('Frequency')
plt.show()

# Perbandingan gaji berdasarkan gender
plt.figure(figsize=(10, 6))
sns.boxplot(x='Gender', y='Salary', data=df)
plt.title('IT Salary Distribution by Gender in EU 2020')
plt.xlabel('Gender')
plt.ylabel('Salary (EUR)')
plt.show()

# Perbandingan gaji berdasarkan negara
plt.figure(figsize=(12, 8))
sns.barplot(x='Country', y='Salary', data=df, estimator=np.median, ci=None)
plt.title('Median IT Salary by Country in EU 2020')
plt.xlabel('Country')
plt.ylabel('Median Salary (EUR)')
plt.xticks(rotation=45, ha='right')
plt.show()

# Korelasi antara gaji, pengalaman, dan pendidikan
plt.figure(figsize=(8, 6))
sns.heatmap(df[['Salary', 'YearsExperience', 'EducationLevel']].corr(), annot=True, cmap='coolwarm', fmt='.2f')
plt.title('Correlation Matrix')
plt.show()
